CHANGELOG.md

```diff
@@ -0,0 +1,289 @@
+- Fixed bugs relating to the `CometLogger` object that would cause it to not work properly
+- Fixed a bug that would occur when returning `-1` from `on_batch_start` following an early exit or when the batch was `None`
+- Fixed a potential race condition with several processes trying to create checkpoint directories
+- Fixed a bug where batch 'segments' would remain on the GPU when using `truncated_bptt > 1`
+- Fixed a bug when using `IterableDataset`
+- Fixed a bug where `.item` was called on non-tensor objects
+- Fixed a bug where `Trainer.train` would crash on an uninitialized variable if the trainer was run after resuming from a checkpoint that was already at `max_epochs`
+- Fixed a bug where early stopping would begin two epochs early
+- Fixed a bug where `num_training_batches` and `num_test_batches` would sometimes be rounded down to zero
+- Fixed a bug where an additional batch would be processed when manually setting `num_training_batches`
+- Fixed a bug when batches did not have a `.copy` method
+- Fixed a bug when using `log_gpu_memory=True` in Python 3.6
+- Fixed a bug where checkpoint writing could exit before completion, giving incomplete checkpoints
+- Fixed a bug where `on_train_end` was not called when ealy stopping
+
+## [0.5.3] - 2019-11-06
+### Added
+- Added option to disable default logger, checkpointer, and early stopping by passing `logger=False`, `checkpoint_callback=False` and `early_stop_callback=False` respectively
+- Added `CometLogger` for use with Comet.ml
+- Added `val_check_interval` argument to `Trainer` allowing validition to be performed at every given number of batches
+- Added functionality to save and load hyperparameters using the standard checkpoint mechanism
+- Added call to `torch.cuda.empty_cache` before training starts
+- Added option for user to override the call t `backward`
+- Added support for truncated backprop through time via the `truncated_bptt_steps` argument in `Trainer` 
+- Added option to operate on all outputs from `training_step` in DDP2
+- Added a hook for modifying DDP init
+- Added a hook for modifying Apex
+### Changed
+- Changed experiment version to be padded with zeros (e.g. `/dir/version_9` becomes `/dir/version_0009`)
+- Changed callback metrics to include any metrics given in logs or progress bar
+- Changed the default for `save_best_only` in `ModelCheckpoint` to `True`
+- Added `tng_data_loader` for backwards compatibility
+- Renamed `MLFlowLogger.client` to `MLFlowLogger.experiment` for consistency
+- Moved `global_step` increment to happen after the batch has been processed
+- Changed weights restore to first attempt HPC weights before restoring normally, preventing both weights being restored and running out of memory
+- Changed progress bar functionality to add multiple progress bars for train/val/test
+- Changed calls to `print` to use `logging` instead
+### Deprecated
+- Deprecated `tng_dataloader`
+### Removed
```

Comment:
just thinking is it would make sense to write there "nothing" to the empty ones so it is clear that nothing changes, not just forgotten

Reply:
Done for Unreleased and last version - should be enough